fluent-plugin-events/lib/sumologic/kubernetes/connector.rb

```diff
@@ -7,17 +7,40 @@ module Connector
       K8_POD_CA_CERT = 'ca.crt'.freeze
       K8_POD_TOKEN = 'token'.freeze
 
-      def connect_kubernetes
-        @client = Kubeclient::Client.new(
-          @kubernetes_url, @api_version,
+      # Need different clients to access different API groups/versions
+      CORE_API_VERSIONS = ['v1'].freeze
+
+      def core_clients
+        CORE_API_VERSIONS.map do |ver|
+          [ver, create_client('api', ver)]
+        end.to_h
+      end
+
+      # If @api_version is not v1, we will create client for other API groups/versions
+      def group_clients
+        [@api_version].map do |ver|
```

Comment:
`@api_version` is default to v1, so this code would create two clients that both connect to v1. we should only create a second client if it the value is not v1

Reply:
The `@api_version` will be checked in `connect_kubernetes`. In that method, when `@api_version` is not equals to `v1`, `group_clients` method won't be called. So there won't be two clients then.
```
def connect_kubernetes
        @clients = @api_version == 'v1' ? core_clients : core_clients.merge(group_clients)
end
```